e2e/node/basic/watermark.test.ts

```diff
@@ -43,115 +44,157 @@ function indexOfQueryResponse(history: Response[]) {
   return history.findIndex(response => response.url.endsWith('query'));
 }
 
-test('basic', async () => {
-  const fetchProxy = new FetchProxy();
-
-  const actor = await createActor({
-    fetch: fetchProxy.fetch.bind(fetchProxy),
-    verifyQuerySignatures: true,
+const watermark1CanisterId =
+  process.env.WATERMARK_CANISTER_ID ?? execSync('dfx canister id watermark1').toString().trim();
+const watermark2CanisterId =
+  process.env.WATERMARK_CANISTER_ID ?? execSync('dfx canister id watermark2').toString().trim();
+const watermark3CanisterId =
+  process.env.WATERMARK_CANISTER_ID ?? execSync('dfx canister id watermark3').toString().trim();
+
+const watermark1Actor = await createActor(watermark1CanisterId);
+const watermark2Actor = await createActor(watermark2CanisterId);
+const watermark3Actor = await createActor(watermark3CanisterId);
+
+describe.sequential('watermark', () => {
+  beforeAll(async () => {
```

Comment:
Same question here, isn't a `beforeEach` enough here?

Reply:
In this case there is only a beforeAll currently, and that is technically doing less since it only runs once. I can switch it though if you prefer beforeEach - it shouldn't be an issue